### ISSUE TITLE
kubectl proxy, API watch request blocks until an event happens

### ISSUE THREAD

**Issue by dhain** [kubernetes/kubectl#628]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.): no

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.): proxy, watch, resourceVersion

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:53:57Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: bare metal
- **OS** (e.g. from /etc/os-release): coreos
- **Kernel** (e.g. `uname -a`): 4.19.25-coreos
- **Install tools**: Typhoon
- **Others**:

**What happened**:
API requests with `watch=true` that don't immediately return any events (eg. with the most recent `resourceVersion`) via `kubectl proxy` completely block without returning HTTP headers until an event comes in.

**What you expected to happen**:
HTTP headers (ie. with `Transfer-Encoding: chunked`) should be returned immediately. This is a problem for my application, because the HTTP library I'm using won't return a streaming response object until it gets the headers from the server.

**How to reproduce it** (as minimally and precisely as possible):
```
$ kubectl proxy &
Starting to serve on 127.0.0.1:8001
$ RESOURCE_VERSION=$(curl -s -XGET http://localhost:8001/api/v1/pods | awk -F\" '/resourceVersion/ { print $4 }' | head -n1)
$ curl --no-buffer --http1.1 -i -XGET "http://localhost:8001/api/v1/pods?resourceVersion=${RESOURCE_VERSION}&watch=true"
```
Above just hangs without showing any output. When I request against the API directly without going through the proxy, it works correctly:
```
$ curl --no-buffer --http1.1 --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" --no-buffer -i -XGET "https://kubernetes.default.svc/api/v1/pods?resourceVersion=${RESOURCE_VERSION}&watch=true"
HTTP/1.1 200 OK
Content-Type: application/json
Date: Wed, 27 Mar 2019 15:52:49 GMT
Transfer-Encoding: chunked

```

**Anything else we need to know**:
This appears to be new behavior in 1.14.0, as this application worked with prior versions of kubectl.

**Comment by dhain** [comment]
Just verified that it works as expected with 1.13.5:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:26:52Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
$ curl --no-buffer --http1.1 -i -XGET "http://localhost:8001/api/v1/pods?resourceVersion=${RESOURCE_VERSION}&wat
ch=true"
HTTP/1.1 200 OK
Content-Type: application/json
Date: Wed, 27 Mar 2019 15:58:15 GMT
Transfer-Encoding: chunked

```

**Comment by dhain** [comment]
Looks like this may have been related to https://github.com/kubernetes/kubernetes/issues/75837 and it appears to be fixed in v1.14.1.
